src/plugins/hetero/src/compiled_model.cpp

```diff
@@ -0,0 +1,762 @@
+                OPENVINO_THROW("Hetero device used default fallback policy, but some layers eg: \n(Name:",
+                               node->get_friendly_name(),
+                               ", Type: ",
+                               node->get_type_name(),
+                               ") were not able to be assigned on any pointed device.\n",
+                               "It happened because these layers are not supported in plugins by default.\n",
+                               "You need to implement custom layers to support them.");
+            } else {
+                OPENVINO_THROW("Network passed to CompiledModel has affinity assigned, but some layers eg: \n(Name:",
+                               node->get_friendly_name(),
+                               ", Type: ",
+                               node->get_type_name(),
+                               ") were not assigned to any device.\n",
+                               "It might happen if you assigned layers manually and missed some layers or\n",
+                               "if you used some automatic assigning mode which decided that these layers are not\n",
+                               "supported by any plugin");
+            }
+        }
+
+        if (dumpDotFile) {
+            ov::hetero::debug::dump_affinities(model, queryNetworkResult, devices);
+        }
+
+        NodeMap<InputSet> nodeInputDependencies;
+        NodeSet graphInputNodes;
+        InputSet subgraphInputs;
+        // Get all subgraph inputs using just node affinities. Also collect transitive closure
+        for (auto&& node : orderedOps) {
+            if (ov::op::util::is_parameter(node) || ov::op::util::is_constant(node)) {
+                graphInputNodes.insert(node.get());
+                subgraphInputs.insert(Input{node.get(), 0});
+                nodeInputDependencies[node.get()].insert(Input{node.get(), 0});
+            } else {
+                auto inputs = node->inputs();
+                auto& nodeInputDependency = nodeInputDependencies[node.get()];
+                for (auto&& input : inputs) {
+                    nodeInputDependency.insert(input);
+                    auto& inputDependency = nodeInputDependencies[InputNode(input)];
+                    nodeInputDependency.insert(inputDependency.begin(), inputDependency.end());
+                    if (affinities[node.get()] != affinities[InputNode(input)]) {
+                        subgraphInputs.insert(input);
+                    }
+                }
+            }
+        }
+
+        // Assign each node subgraph ID
+        auto CollectSubgraphs = [&] {
+            std::deque<int> subgraphIds;
+            NodeMap<int*> subgraphIdPtrs;
+            for (auto&& node : orderedOps) {
+                auto allNodeInputs = node->inputs();
+                std::vector<Input> inputs;
+                for (auto&& input : allNodeInputs) {
+                    if (!InferenceEngine::details::contains(subgraphInputs,
```

Comment:
Why do you use legacy API?

Reply:
Added new implementation of `ov::util::contains()` (copy of `InferenceEngine::details::contains`) for `set` and other containers

Reply:
https://github.com/openvinotoolkit/openvino/pull/18429/files#r1259449730